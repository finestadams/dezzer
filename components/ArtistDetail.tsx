import { useAppSelector } from "app/hooks";
import React, { useEffect, useState } from "react";
import AlbumList from "./AlbumList";
import PlayLists from "./PlayLists";
import Images from "next/image";

const ArtistDetail = ({ albumDetail }: any) => {
  const artistImg = useAppSelector((state) => state.artist.currentArtistImage);
  const artistName = useAppSelector((state) => state.artist.currentArtistName);
  const noOfFans = useAppSelector((state) => state.artist.currentNoOfFans);
  const getArtistDetails = albumDetail?.data?.map((data: any) => {
    return (
      <>
        <PlayLists data={data} key={data.id}></PlayLists>
      </>
    );
  });
  const getAlbumDetails = albumDetail?.data?.map((data: any) => {
    return (
      <>
        <AlbumList data={data} key={data.id}></AlbumList>
      </>
    );
  });

  return (
    <>
      <div className="flex flex-col justify-center md:flex-row md:max-w-1/2 relative">
        <img
          className="object-cover w-full h-96 rounded-t-lg md:h-72 md:w-1/2 md:rounded-none "
          src={artistImg}
          alt="img"
        />
        <div className="absolute left-10 bottom-1/2 ">
          <h1 className="font-semibold text-2xl ">{artistName}</h1>
          <p className="text-small font-light">{noOfFans} fans</p>
        </div>
        <div className="flex flex-col justify-start p-4 leading-normal flex-1">
          <h5 className=" text-2xl font-bold tracking-tight text-gray-900 dark:text-white mb-5">
            Top tracks
          </h5>
          {getArtistDetails}{" "}
        </div>
      </div>
      <div className="mt-4 ">
        <h2 className="grid grid-cols-1 gap-3 w-3/4 m-auto md:grid-cols-3 md:gap-4 md:w-4/6 md:m-auto text-2xl py-4">
          Albums
        </h2>
        <div className="grid grid-cols-1 gap-3 w-3/4 m-auto md:grid-cols-3 md:gap-4 md:w-4/6 md:m-auto">
          {getAlbumDetails}
        </div>
      </div>
    </>
  );
};

export default ArtistDetail;
