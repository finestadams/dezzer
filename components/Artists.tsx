import { useAppDispatch } from "app/hooks";
import React, { useEffect, useState } from "react";
import ArtistDetailService from "services/getartistdetail";
import {
  artistImage,
  artistName,
  artistTrackId,
  noOfFans,
} from "features/artists/artistSlice";
import { useRouter } from "next/router";

const Artists = ({ getAllArtists }: any) => {
  const [mergeData, setMergData] = useState<any[]>([]);
  const dispatch = useAppDispatch();
  const router = useRouter();
  const getArstistDetail = async () => {
    const data = getAllArtists.data.map(async (data: any) => {
      let artistResponse = await ArtistDetailService.getArtistDetail(data.id);
      if (artistResponse.status === 200) {
        const resp = artistResponse.data;
        return {
          ...data,
          resp,
        };
      } else {
        return data;
      }
    });
    const finalData = await Promise.all(data);
    setMergData(finalData);
  };
  useEffect(() => {
    getArstistDetail();
  }, []);

  const eachArtistAfterMap = mergeData.map((data: any) => {
    return (
      <div
        className="rounded-lg shadow-lg bg-white max-w-sm cursor-pointer"
        key={data.id}
        onClick={() => {
          dispatch(artistTrackId(data?.resp?.id));
          dispatch(artistImage(data?.picture_big));
          dispatch(noOfFans(data?.resp?.nb_fan));
          dispatch(artistName(data.name));
          router.replace(`/artists/${data?.resp?.id}`);
        }}
      >
        <img className="rounded-t-lg" src={data?.picture_big} alt="" />
        <div className="p-6">
          <h5 className="text-gray-900 text-xl font-medium mb-2">
            {data?.name}
          </h5>
          <p className="text-gray-700 text-base mb-4">
            {data?.resp?.nb_fan} fans
          </p>
        </div>
      </div>
    );
  });
  return <>{eachArtistAfterMap}</>;
};

export default Artists;
