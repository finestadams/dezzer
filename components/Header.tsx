import { useAppDispatch } from "app/hooks";
import { searchArtist } from "features/artists/artistSlice";
import Link from "next/link";
import React, { useEffect, useState } from "react";

const Header = () => {
  const [search, setSearch] = useState("");
  const [localStorageExists, setLocalStorageExists] = useState(false);
  const dispatch = useAppDispatch();

  useEffect(() => {
    if (!localStorage.getItem("favourites-album")) {
      setLocalStorageExists(true);
    }
  }, [localStorageExists]);
  console.log(typeof localStorage.getItem("favourites-album"), "to");
  useEffect(() => {
    if (!search) {
      dispatch(searchArtist(search));
    }
  }, []);
  return (
    <>
      <nav className="bg-white border-gray-200 px-2 sm:px-4 py-5 rounded dark:bg-gray-900 shadow">
        <div className="container flex space-x-9  items-center mx-auto">
          <div>
            <Link href="/artists">
              <a className="flex items-center">
                <img
                  src="/svg/logo.svg"
                  className="mr-3 h-6 sm:h-9"
                  alt="Deezer Logo"
                />
              </a>
            </Link>
          </div>

          <div
            className=" w-full md:block md:w-1/2 relative"
            id="navbar-default"
          >
            <form>
              <input
                type="search"
                className="form-control block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:outline-none"
                id="exampleSearch"
                placeholder="Search for artists"
                value={search}
                onChange={(e) => {
                  setSearch(e.target.value);
                  setTimeout(() => {
                    dispatch(searchArtist(search));
                  }, 5000);
                }}
              />
            </form>
          </div>
          <div>
            <Link href={!localStorageExists ? "/album/favourites" : "#"}>
              <a className="flex items-center">Favourites</a>
            </Link>
          </div>
        </div>
      </nav>
    </>
  );
};

export default Header;
