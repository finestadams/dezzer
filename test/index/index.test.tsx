import { render, screen } from "@testing-library/react";
import ArtistDetail from "components/ArtistDetail";
import Artists from "components/Artists";
import Header from "components/Header";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import { expect } from "@jest/globals";
import Footer from "components/Footer";

describe("Artists", () => {
  let store: any;
  it("Artist component should render", () => {
    <Provider store={store}>
      <Artists />
    </Provider>;
  });
  it("Artist detail component should render", () => {
    <Provider store={store}>
      <ArtistDetail />
    </Provider>;
  });
});

describe("Footert", () => {
  let store: any;
  it("footer", () => {
    render(<Footer />);
  });
  const { container } = render(<Footer />);
  expect(container).toMatchSnapshot();
});
