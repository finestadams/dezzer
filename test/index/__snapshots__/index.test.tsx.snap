// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <div
    class="p-4 bg-white md:px-6 md:py-8 "
  >
    <hr
      class="my-6 border-gray-200 sm:mx-auto dark:border-gray-700 lg:my-8"
    />
    <span
      class="block text-sm text-gray-500 text-center"
    >
      Powered by
       
      <a
        class="hover:underline text-black text-[14px]"
        href="/"
        rel="noreferrer"
        target="_blank"
      >
        Deezer
      </a>
    </span>
  </div>
</div>
`;
